lib/java/desktop-environment/src/main/java/org/enso/desktopenvironment/directories/WindowsDirectories.java

```diff
@@ -0,0 +1,48 @@
+package org.enso.desktopenvironment.directories;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.concurrent.TimeUnit;
+
+class WindowsDirectories implements Directories {
+
+  private static final String[] PROCESS_REG_QUERY =
+      new String[] {
+        "reg",
+        "query",
+        "\"HKCU\\Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\ShellFolders\"",
+        "/v",
+        "personal"
+      };
+
+  /**
+   * Get the path to 'My Documents' user directory.
+   *
+   * <p>Method uses the registry query that may not work on Windows XP versions and below.
+   *
+   * @return the 'My Documents' user directory path.
+   * @throws DirectoriesException when fails to detect the user documents directory.
+   */
+  @Override
+  public Path getDocuments() throws DirectoriesException {
+    try {
+      var process = new ProcessBuilder(PROCESS_REG_QUERY).start();
```

Comment:
Avoiding a process execution would be nicer, but not even [stack over flow has a suggestion](https://stackoverflow.com/questions/62289/read-write-to-windows-registry-using-java) how to do that nicely.